frontend/src/lib/components/chart/ChartHomeBlock.svelte

```diff
@@ -69,10 +69,17 @@
 										name: 'Copy of ' + chart.name,
 										type: chart.type,
 										parameters: chart.parameters
-									}).then(() => {
-										ZenoService.getCharts($project ? $project.uuid : '').then((fetchedCharts) =>
-											charts.set(fetchedCharts)
-										);
+									}).then((res) => {
+										invalidateAll();
+										charts.update((c) => {
+											c.push({
+												id: res,
+												name: 'Copy of ' + chart.name,
+												type: chart.type,
+												parameters: chart.parameters
+											});
+											return c;
+										});
```

Comment:
I fetched after updates to make sure we have the most recent result. Don't know if that's a big problem but this might be an issue with concurrent updates. Can probably merge for now though.